UI/signIn.tsx

```diff
@@ -2,18 +2,23 @@
 import { Component, h } from "https://esm.sh/preact@10.17.1";
 import { tw } from "https://esm.sh/twind@0.16.16";
 import { GetLocalStorageAccountContext, Nip7ExtensionContext } from "./account-context.ts";
-import { ButtonClass, CenterClass, DividerClass } from "./components/tw.ts";
+import { ButtonClass, CenterClass, LinearGradientsClass } from "./components/tw.ts";
 import KeyView from "./key-view.tsx";
 import { PrivateKey } from "../lib/nostr-ts/key.ts";
-import { InMemoryAccountContext } from "../lib/nostr-ts/nostr.ts";
-import { emitFunc, EventEmitter } from "../event-bus.ts";
+import { InMemoryAccountContext, NostrAccountContext } from "../lib/nostr-ts/nostr.ts";
+import { emitFunc } from "../event-bus.ts";
+import {
+    ErrorColor,
+    HintTextColor,
+    HoverButtonBackgroudColor,
+    PrimaryBackgroundColor,
+    PrimaryTextColor,
+    SecondaryBackgroundColor,
+} from "./style/colors.ts";
 
 export type SignInEvent = {
-    type: "signin";
-    privateKey?: PrivateKey; // undefined means sign in with extentions
-} | {
-    type: "editSignInPrivateKey";
-    privateKey: string;
+    type: "SignInEvent";
+    ctx: NostrAccountContext;
```

Comment:
Good practice on Type Level programming. We only expose necessary data/interfaces to downstream